plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,662 @@
+In the first period of the exit, any user may reveal a competing transaction that potentially makes the exiting transaction non-canonical.
+This competing transaction must be exitable and must share an input with the exiting transaction, but does not have to be included in the Plasma chain.
+Multiple competing transactions can be revealed during this period, but only the oldest presented transaction is considered for the purposes of a response.
+
+If any transactions have been presented during the first period, any other user can respond to the challenge by proving that the exiting transaction is actually included in the chain before the oldest presented competing transaction.
+If this response is given before the end of the second period, then the exiting transaction is determined to be canonical and the responder receives the `exit bond` placed by the user who started the exit.
+Otherwise, the exiting transaction is determined to be non-canonical and the challenger receives `exit bond`.
+
+Note that this challenge means it’s possible for an honest user to lose `exit bond` as they might not be aware their transaction is non-canonical.
+We address this attack vector and several mitigations in detail later.
+
+<!-- TODO: Include image of canonicity "state machine" -->
+
+
+#### Piggybacking an Exit
+
+As noted earlier, it’s possible that some participants in a transaction may not be aware that the transaction is non-canonical.
+Owners of both inputs and outputs to a transaction may want to start an exit in the case that they would receive the funds from the exit.
+However, we want to avoid the gas cost of repeatedly publishing and proving statements about the same transaction.
+We therefore allow owners of inputs or outputs to a transaction to piggyback an existing exit that references the transaction.
+
+Users must piggyback an exit within the first period.
+To piggyback an exit, an input or output owner places a bond, `piggyback bond`.
+This bond is used to cover the cost of challenges that show the input or output is spent.
+A successful challenge blocks the specified input or output from exiting.
+These challenges must be presented before the end of the second period.
+
+Note that it isn’t mandatory to piggyback an exit.
+Users who choose not to piggyback an exit are choosing not to attempt a withdrawal of their funds.
+If the chain is byzantine, not piggybacking could potentially mean loss of funds.
+
+
+#### Combining with Plasma MVP Exit Protocol
+
+The MoreVP protocol can be combined with the Plasma MVP protocol in a way that simultaneously preserves the integrity of exits and minimizes gas cost. 
+Owners of outputs on the Plasma chain should be able to start an exit via either mechanism, but not both.
+Although the two protocols use different determinations for exit priority, we still need a total ordering on exits.
+Therefore, every exit, no matter the protocol used, must be included in the same priority queue for processing.
+
```

Comment:
Is there any scenario where using youngest input as the priority criterion for a "standard" exit creates a problem?

Reply:
If you'd do that, i.e. have youngest input priority but without the in-flight exit protocol (exit game), the operator would be able to multi-spend an old output and exit all of the multi spending transactions, with higher priority than "honest exits".

With the more complex/costly MoreVP exit protocol, there is no such possibility.

Reply:
But let's say we leave the in-flight exit protocol as-is.  The operator will be challenged for producing multiple competing exit transactions, and all will be rolled back at operator's expense.

I'm just looking for a way to reconcile the ordering system.

I'm trying to grasp the fundamental difference between the "standard" and "moreVP" exit protocols, and whether it's true that the former requires a different priority ordering (tx # vs. youngest input) to work properly.  If not, then combining them into the same queue becomes trivial.

Reply:
They use a different priority ordering but they can still be combined into the same queue.

Reply:
@jepidoptera I think the sequence of implications is for

> "standard" and "moreVP" exit protocols, and whether it's true that the former requires a different priority ordering (tx # vs. youngest input)

is this: moreVP protocol is expensive => we keep the standard exit => standard exit uses output position for ordering => we don't care about inputs to that transaction having been valid or not

or the other way around: in-flight transaction's outputs may have potentially low priority => in-flight transaction's outputs are doomed in standard exit proto => need to use their inputs as "source of higher priority" => need to prove more => more expensive, moreVP protocol for in-flight txs.

This is how I got the hang of that difference, does this help?